app/src/androidTest/java/com/example/evently/ViewNotificationsTest.java

```diff
@@ -0,0 +1,251 @@
+import com.example.evently.data.model.Event;
+import com.example.evently.data.model.Notification;
+import com.example.evently.ui.entrant.ViewNotificationsFragment;
+
+@RunWith(AndroidJUnit4.class)
+public class ViewNotificationsTest extends EmulatedFragmentTest<ViewNotificationsFragment> {
+    private static final EventsDB eventsDB = new EventsDB();
+    private static final NotificationDB notificationDB = new NotificationDB();
+
+    private static final Instant now = Instant.now();
+    // We can use the same times for these tests.
+    private static final Timestamp selectionTime = new Timestamp(now.plus(Duration.ofMillis(100)));
+    private static final Timestamp eventTime = new Timestamp(now.plus(Duration.ofMinutes(10)));
+
+    // Create a few events.
+    private static final Event[] mockEvents = new Event[] {
+        new Event(
+                "name",
+                "description",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name1",
+                "description1",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name2",
+                "description2",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name3",
+                "description3",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name4",
+                "description4",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name5",
+                "description5",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name6",
+                "description6",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name7",
+                "description7",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50),
+        new Event(
+                "name8",
+                "description8",
+                Category.EDUCATIONAL,
+                selectionTime,
+                eventTime,
+                "orgEmail",
+                50)
+    };
+
+    @BeforeClass
+    public static void setUpNotifications() {
```

Comment:
Any full-test-long data you need, set it in a `@BeforeClass` method which must be static.

For these tests, I obviously need a pre existing set of events and I have to enroll in some of them. So I do that here.